controllers/workspace/status.go

```diff
@@ -238,3 +240,29 @@ func updateMetricsForPhase(workspace *dw.DevWorkspace, oldPhase, newPhase dw.Dev
 		metrics.WorkspaceStarted(workspace, logger)
 	}
 }
+
+// checkForStartTimeOut checks if the provided workspace has not progressed for longer than the configured
+// startup timeout. This is determined by checking to see if the last condition transition time is more
+// than [timeout] duration ago. Workspaces that are not in the "Starting" phase cannot timeout. Returns
+// an error with message when timeout is reached.
+func checkForStartTimeOut(workspace *dw.DevWorkspace, logger logr.Logger, status *currentStatus) error {
+	if workspace.Status.Phase != dw.DevWorkspaceStatusStarting {
+		return nil
+	}
+	timeout, err := time.ParseDuration(config.Workspace.StartProgressTimeout)
```

Comment:
Parsing once would mean it's harder to pick up changes in config while running (we'd probably have to restart the controller).

We could avoid this by using the `Duration` raw in the config, since "parsing" is just converting a string to a `int64`, but that would require the entered value to be in nanoseconds (I believe)

Reply:
> Parsing once would mean it's harder to pick up changes in config while running (we'd probably have to restart the controller).

True, but also we can reevaluate that value on DWOC reconciliation. but don't mind, that's good enough or even better than I have in mind =)